diff --git a/core/SolverTypes.h b/core/SolverTypes.h
index 3a93f05..56cd152 100644
--- a/core/SolverTypes.h
+++ b/core/SolverTypes.h
@@ -42,12 +42,14 @@ namespace Minisat {
 typedef int Var;
 #define var_Undef (-1)
 
+struct Lit;
+inline Lit mkLit(Var var, bool sign = false);
 
 struct Lit {
     int     x;
     
     // Use this as a constructor:
-    friend Lit mkLit(Var var, bool sign = false);
+    friend Lit mkLit(Var var, bool sign);
 
     bool operator == (Lit p) const { return x == p.x; }
     bool operator != (Lit p) const { return x != p.x; }
